CONTRIBUTING.md

```diff
@@ -93,94 +115,194 @@ Here is a great example of a well defined issue: https://github.com/angular/angu
-  that relates to your submission. You don't want to duplicate effort.
-* Please sign our [Contributor License Agreement (CLA)](#cla) before sending pull
-  requests. We cannot accept code without this.
-* Make your changes in a new git branch:
+## <a name="contributions"></a> Code and Documentation Contributions
 
-    ```shell
-    git checkout -b my-fix-branch master
-    ```
+### Development Setup
 
-* Create your patch, **including appropriate test cases**.
-* Follow our [Coding Rules](#rules).
-* Run the full AngularJS test suite, as described in the [developer documentation][dev-doc],
-  and ensure that all tests pass.
-* Commit your changes using a descriptive commit message that follows our
-  [commit message conventions](#commit) and passes our commit message presubmit hook
-  (`validate-commit-msg.js`). Adherence to the [commit message conventions](#commit) is required,
-  because release notes are automatically generated from these messages.
+This document describes how to set up your development environment to build and test AngularJS, and
+explains the basic mechanics of using `git`, `node`, `yarn`, `grunt`, and `bower`.
 
-    ```shell
-    git commit -a
-    ```
-  Note: the optional commit `-a` command line option will automatically "add" and "rm" edited files.
+#### Installing Dependencies
 
-* Build your changes locally to ensure all the tests pass:
+Before you can build AngularJS, you must install and configure the following dependencies on your
+machine:
 
-    ```shell
-    grunt test
-    ```
+* [Git](http://git-scm.com/): The [Github Guide to
+  Installing Git](https://help.github.com/articles/set-up-git) is a good source of information.
 
-* Push your branch to GitHub:
+* [Node.js v6.x (LTS)](http://nodejs.org): We use Node to generate the documentation, run a
+  development web server, run tests, and generate distributable files. Depending on your system,
+  you can install Node either from source or as a pre-packaged bundle.
 
-    ```shell
-    git push origin my-fix-branch
-    ```
+  We recommend using [nvm](https://github.com/creationix/nvm) (or
+  [nvm-windows](https://github.com/coreybutler/nvm-windows))
+  to manage and install Node.js, which makes it easy to change the version of Node.js per project.
+
+* [Yarn](https://yarnpkg.com): We use Yarn to install our Node.js module dependencies
+  (rather than using npm).
+  There are detailed installation instructions available at https://yarnpkg.com/en/docs/install.
+
+* [Java](http://www.java.com): We minify JavaScript using
+  [Closure Tools](https://developers.google.com/closure/), which require Java (version 7 or higher)
+  to be installed and included in your
+  [PATH](http://docs.oracle.com/javase/tutorial/essential/environment/paths.html) variable.
+
+* [Grunt](http://gruntjs.com): We use Grunt as our build system. Install the grunt command-line tool
+  globally with:
+
+  ```shell
+  yarn global add grunt-cli
```

Comment:
Wouldn't it be a better idea to avoid global installation of grunt by default ?
People unfamiliar with yarn/npm will follow the guideline and install it globally. I think people should NOT install this globally unless they know why.

Reply:
As discussed in https://github.com/angular/angular.js/pull/16276, this is something we can do: document `yarn ...` as the default and list global installing of grunt as an alternative. But I think this can be part of a follow-up PR, as there may be more files that are affected by this.